core/model/asynctask.py

```diff
@@ -0,0 +1,115 @@
+# Async
+import datetime
+import json
+import uuid
+from enum import Enum
+from typing import Any
+
+from pydantic.dataclasses import dataclass
+from sqlalchemy import Column, DateTime, String
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy.types import Enum as SqlAlchemyEnum
+
+from core.model import Base, create
+from core.util.datetime_helpers import utc_now
+
+
+class AsyncTaskStatus(str, Enum):
+    READY = "READY"
+    PROCESSING = "PROCESSING"
+    SUCCESS = "SUCCESS"
+    FAILURE = "FAILURE"
+
+
+class AsyncTaskType(Enum):
+    INVENTORY_REPORT = "INVENTORY_REPORT"
+
+
+class AsyncTask(Base):
+    """An asynchronous task."""
+
+    __tablename__ = "asynctasks"
```

Comment:
Do we need a job of some sort to clear out old entries in this table? Maybe tasks that has finished more then a month ago or something like that. It seems like it might be a bit of a problem to have this table continue to grow forever.

Reply:
I'll add it in.